### ISSUE TITLE
[Bug] Wrong assertion when dragging a text node

### ISSUE THREAD

**Issue by boris-petrov** [emberjs/ember.js#20610]
### 🐞 Describe the Bug
Dragging text instead of an HTML element leads to an assertion error. See below in the reproduction.

### 🔬 Minimal Reproduction
Having this HTML:
```html
Some Text <a href="#">Some Link</a>
```

Select everything, hover the text and try dragging it. There will be an assertion failure.

The problem is [here](https://github.com/emberjs/ember.js/blob/v5.5.0/packages/%40ember/-internals/views/lib/system/event_dispatcher.ts#L319). Perhaps the check `target instanceof Element` is overly restrictive.

### 😕 Actual Behavior
An assertion which is wrong.

### 🤔 Expected Behavior
No assertion.

### 🌍 Environment

- Ember: 5.5.0
- Node.js/npm: v21.4.0
- OS: Linux
- Browser: Chromium

**Comment by johanrd** [comment]
@boris-petrov I tried to reprocuce now without luck. Are you still able to? thanks.

**Comment by boris-petrov** [comment]
@johanrd yes, the same problem still occurs in Ember 7.1.0. The [code](https://github.com/emberjs/ember.js/blob/v7.1.0-ember-source/packages/%40ember/-internals/views/lib/system/event_dispatcher.ts#L270) is still the same.

Make sure you do EXACTLY what I've written - select both text AND link, hover over the TEXT, drag. You'll see `Assertion Failed: [BUG] Received event without an Element target: dragstart, [object Text]`.

**Comment by johanrd** [comment]
Hmm, i think I did. Still cannot reproduce, maybe i do it wrongly:/ Can you perhaps show by video? 

https://github.com/user-attachments/assets/e27798e8-5395-4167-a6cb-352178034068

**Comment by boris-petrov** [comment]
Thanks for taking the time! So what was needed (unknown to me) was an Ember component that registers an event:

`components/some-component.ts`
```ts
import Component from '@ember/component';

export default class SomeComponent extends Component {
  dragStart() {
  }
}
```

`templates/application.hbs`
```hbs
<SomeComponent />
Some Text <a href="#">Some Link</a>
```

And do the same - select all, drag the text.

I can push a repo if you need.

**Comment by johanrd** [comment]
okay, thanks, i can reproduce now with classic `@ember/component` but not with `@glimmer/component`.

There is an open RFC to deprecate classic '@ember/component' (https://github.com/emberjs/rfcs/pull/1216), so maybe not a high priority fix.

**Comment by boris-petrov** [comment]
Absolutely. If it requires an Ember component and the broken code is not called from anything else - then yes, this bug can safely be ignored.
